File: 2810-faulty-keyboard/2810-faulty-keyboard.js
/**
 * @param {string} s
 * @return {string}
 */
var finalString = function(s) {
    let split = s.split('')
    let result = '';
    
    for (let i = 0; i < split.length; i++) {
        if (split[i] === 'i') {
            result = result.split('').reverse().join('')
        } else {
            result += split[i];
        }
    }
    return result;
};

File: 2810-faulty-keyboard/2810-faulty-keyboard.cs
public class Solution {
    public string FinalString(string s) {
        string result = "";
        foreach(var letter in s) {
            if(letter == 'i') {
                string reversedString = new string(result.ToCharArray().Reverse().ToArray());
                result = reversedString;
            }
            else {
                result += letter;
            }
        }
        return result;
    }
}